direct.go

```diff
@@ -257,14 +308,34 @@ func (d *direct) RoundTrip(req *http.Request) (*http.Response, error) {
 			log.Debugf("Retrying domain-fronted request, pass %d", i)
 		}
 
-		req.Body = getBody()
-		conn, masqueradeGood, err := d.dial()
+		conn, m, masqueradeGood, err := d.dial()
 		if err != nil {
 			// unable to find good masquerade, fail
 			return nil, err
 		}
+		provider := d.providerFor(m)
+		if provider == nil {
+			log.Debugf("Skipping masquerade with disabled/unknown provider '%s'", m.ProviderID)
+			masqueradeGood(false)
+			continue
+		}
+		frontedHost := provider.Lookup(originHost)
+		if frontedHost == "" {
+			log.Tracef("Not translating unknown origin %s...", originHost)
+			frontedHost = originHost
```

Comment:
It shouldn't ... it could be an error instead if I adjust how the test urls are handled I think.

Reply:
I would argue that if you're trying to front to a host for which you don't have an alias, it should be an error and fail hard.

Reply:
It's a bit funky in the general case, but if I validate early that providers cover the same set of hosts, I think it could be considered a caller error and not affect the validity of the masquerade?

Reply:
> if I validate early that providers cover the same set of hosts

You could do that, but I don't think it's strictly necessary as long as we're handling this case correctly here. In practice, whatever scripts we use for generating the provider config can handle making sure that we have a full set of aliases for each provider.